tests/flytekit/unit/extras/tasks/test_shell.py

```diff
@@ -99,62 +140,144 @@ def test_input_output_substitution_files():
         assert s == contents
 
 
-def test_input_single_output_substitution_files():
-    s = """
-        cat {{ .inputs.f }} >> {{ .outputs.y }}
-        echo "Hello World {{ .inputs.y }} on  {{ .inputs.j }}"
-        """
+@pytest.mark.parametrize(
+    "template_style,script,output_location",
+    [
+        (
+            "default",
+            """
+            cat {{ .inputs.f }} >> {{ .outputs.z }}
```

Comment:
example where I had to change the output variable `y` since it was the same as in the input. Did this for ease of implementation in this POC